src/main/java/com/novi/springboottechiteasy/services/CiModuleService.java

```diff
@@ -0,0 +1,124 @@
+
+            return ciModuleDtoMapper.mapToDto(foundModule);
+        } else {
+            throw new RecordNotFoundException("No remote-controller found with id: " + id);
+        }
+    }
+
+    public CiModuleDto addCiModule(CiModuleInputDto inputDto) {
+        CiModule ciModule = transferToCiModule(inputDto);
+        ciModuleRepository.save(ciModule);
+        return ciModuleDtoMapper.mapToDto(ciModule);
+    }
+
+    public CiModuleDto updateCiModule(Long id, CiModuleInputDto newCiModule) {
+        Optional<CiModule> ciModule = ciModuleRepository.findById(id);
+
+        if (ciModule.isPresent()) {
+
+            CiModule thisCiModule = ciModule.get();
+            thisCiModule.setBrand(newCiModule.getBrand());
+            thisCiModule.setType(newCiModule.getType());
+            thisCiModule.setPrice(newCiModule.getPrice());
+
+            CiModule saveCiModule = ciModuleRepository.save(thisCiModule);
+
+            return ciModuleDtoMapper.mapToDto(saveCiModule);
+        } else {
+            throw new RecordNotFoundException("No remote-controller found with id: " + id);
+        }
+    }
+
+    public void deleteCiModule(Long id) {
+        ciModuleRepository.deleteById(id);
+    }
+
+    public CiModuleDto updateCiModuleDetails(Long id, CiModuleInputDto updatedCiModule) {
+        Optional<CiModule> ciModule = ciModuleRepository.findById(id);
+
+        if (ciModule.isPresent()) {
+
+            CiModule thisCiModule = ciModule.get();
+            if (updatedCiModule.getBrand() != null) {
+                thisCiModule.setBrand(updatedCiModule.getBrand());
+            }
+            if (updatedCiModule.getType() != null) {
+                thisCiModule.setType(updatedCiModule.getType());
+            }
+            if (updatedCiModule.getPrice() != null) {
+                thisCiModule.setPrice(updatedCiModule.getPrice());
+            }
+
+            CiModule saveCiModule = ciModuleRepository.save(thisCiModule);
+
+            return ciModuleDtoMapper.mapToDto(saveCiModule);
+        } else {
+            throw new RecordNotFoundException("No remote-controller found with id: " + id);
+        }
+    }
+
+    public CiModule transferToCiModule(CiModuleInputDto inputDto) {
+        CiModule ciModule = new CiModule();
+
+        ciModule.setBrand(inputDto.getBrand());
+        ciModule.setType(inputDto.getType());
+        ciModule.setPrice(inputDto.getPrice());
+
+        if (inputDto.getCompatibleTelevisionIds() != null) {
+            List<Television> compatibleTelevisions = new ArrayList<>();
+            for (Long televisionId : inputDto.getCompatibleTelevisionIds()) {
+                Television television = televisionRepository.findById(televisionId)
+                        .orElseThrow(() -> new RecordNotFoundException("No television found with id: " + televisionId));
+                compatibleTelevisions.add(television);
+            }
+            ciModule.setCompatibleTelevisions(compatibleTelevisions);
```

Comment:
Het heeft weinig nut om de compatibleTevisions van CiModule te zetten, omdat CiModule niet de owner is van de relatie.

Reply:
Door je gebruik van cascade zou dit weleens onwaar kunnen zijn